curl/tests/client_test.cc

```diff
@@ -487,15 +491,30 @@ namespace leatherman { namespace curl {
              REQUIRE_THROWS_WITH(test_client.download_file(req, file_path), "easy perform failed"); 
 
              // Ensure that the temp file was removed
-             REQUIRE_THROWS_AS(find_matching_file(TEMP_FILE_REGEX), std::runtime_error);
+             REQUIRE(fs::is_empty(temp_dir_path));
          }
 
-         SECTION("when curl_easy_perform fails and removal of the temporary file fails, then both the errbuf and the failed temporary file removal message are contained in the thrown http_file_download_exception") {
+         SECTION("when curl_easy_perform fails and removal of the temporary file fails, then both the errbuf and the failed temporary file removal message are contained in the thrown http_file_operation_exception") {
              std::string file_path = (temp_dir_path / "file").string();
              curl::request req("");
              test_impl->test_failure_mode = curl_impl::error_mode::easy_perform_error; 
              test_impl->trigger_external_failure = setup_temp_removal_failure;
              REQUIRE_THROWS_WITH(test_client.download_file(req, file_path), Catch::StartsWith("easy perform failed and failed to remove temporary file file_util_fixture_")); 
          }
+
+         SECTION("when curl_easy_perform fails due to a CURLE_WRITE_ERROR, an http_file_operation_exception is thrown") {
```

Comment:
In the past, I think I've done the try-catch and check for message. You have to have a REQUIRE(false) at the end of the try to ensure you don't reach that point.